pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go

```diff
@@ -114,6 +111,19 @@ type RaftNode interface {
 	// NB: NextUnstableIndex can regress when the node accepts appends or
 	// snapshots from a newer leader.
 	NextUnstableIndexLocked() uint64
+	// ReplicasStateLocked returns the current status state of all replicas. The
+	// value of Match, Next are populated iff in StateReplicate. All entries >=
```

Comment:
These values are always populated. E.g. in `StateProbe` they reflect the position at which raft is probing the appends. Maybe we don't need this detail here - is it enough to know that we will only be **using** the values if state is `StateReplicate`?

Reply:
I think that is fine judging by the prototype code and what we have so far.

Reply:
I mean, this comment seems inaccurate? The implementation of this method does not condition `Match/Next` field population to `StateReplicate`, and neither does `RawNode`. I.e. these are not zero values in non-`StateReplicate` case. In fact, `Match` is always accurate, and `Next` is always > `Match`.

> I think that is fine

Do you mean it's fine that these fields are populated - we don't use `Match/Next` if not in `StateReplicate`? Should the comment say this much? E.g.:

```golang
// ReplicasStateLocked returns the replication flow state for all replicas.
//
// RACv2 uses the Match and Next indices only for StateReplicate replicas.
// ... elaboration on what these mean in StateReplicate ...
//
// ... elaboration on transitions to/from StateReplicate ...
```